Modules/Devices/test/Devices.Application.Tests/Tests/Devices/Commands/UpdateDevice/HandlerTests.cs

```diff
@@ -0,0 +1,109 @@
+﻿using Backbone.BuildingBlocks.Application.Abstractions.Exceptions;
+using Backbone.BuildingBlocks.Application.Abstractions.Infrastructure.UserContext;
+using Backbone.DevelopmentKit.Identity.ValueObjects;
+using Backbone.Modules.Devices.Application.Devices.Commands.UpdateDevice;
+using Backbone.Modules.Devices.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Devices.Application.Tests.Extensions;
+using Backbone.Modules.Devices.Domain.Entities.Identities;
+using Backbone.UnitTestTools.BaseClasses;
+using FakeItEasy;
+using FluentAssertions;
+using Microsoft.Extensions.Logging;
+using Xunit;
+using ApplicationException = Backbone.BuildingBlocks.Application.Abstractions.Exceptions.ApplicationException;
+using Handler = Backbone.Modules.Devices.Application.Devices.Commands.UpdateDevice.Handler;
+
+namespace Backbone.Modules.Devices.Application.Tests.Tests.Devices.Commands.UpdateDevice;
+
+public class HandlerTests : AbstractTestsBase
+{
+    [Fact]
+    public async Task Updates_device_with_values_sent()
+    {
+        // Arrange
+        var activeDevice = CreateDevice();
+
+        var mockIdentitiesRepository = A.Fake<IIdentitiesRepository>();
+        A.CallTo(() => mockIdentitiesRepository.GetDeviceById(activeDevice.Id, A<CancellationToken>._, A<bool>._)).Returns(activeDevice);
+
+        var fakeUserContext = A.Fake<IUserContext>();
+        A.CallTo(() => fakeUserContext.GetDeviceId()).Returns(activeDevice.Id);
+
+        var handler = CreateHandler(mockIdentitiesRepository, fakeUserContext);
+        var updateDeviceCommand = new UpdateDeviceCommand
+        {
+            CommunicationLanguage = "en"
+        };
+
+        // Act
+        await handler.Handle(updateDeviceCommand, CancellationToken.None);
+
+        // Assert
+        activeDevice.CommunicationLanguage.Should().Be(updateDeviceCommand.CommunicationLanguage);
+
+        A.CallTo(() => mockIdentitiesRepository.Update(
+            A<Device>.That.Matches(d => d.CommunicationLanguage == updateDeviceCommand.CommunicationLanguage),
+            A<CancellationToken>._
+        )).MustHaveHappenedOnceExactly();
+    }
+
+    [Fact]
+    public async Task Throws_when_provided_an_invalid_communication_language()
```

Comment:
This test can be deleted once we have extracted the logic into a value object. And remember to create a test for the value object instead.